### PR TITLE
Adds logic to switch_register cli to accomodate disparate inputs

### PR DESCRIPTION
**Pull request by SahilTikale**
Currently supporting mock, dell and nexus

### CODE REVIEW

**Comment by SahilTikale** [comment]
@knikolla @henn 
simple PR, have a look when you can.

**Comment by SahilTikale** [comment]
@kylehogan please have a look if you have time.

**Comment by knikolla** [comment]
-1. Instead of using *args, you should use **kwargs.

**Comment by SahilTikale** [comment]
cli should be taking command line arguments hence args is the appropriate thing. 
I do not wish to use kwargs as it will compels the command line user to know what keys are there for each switch driver. 

About code duplication, I thought about it but I do not think that I can simply it.
I can make the error as a single function that can be invoked if desired input is not received, but that would make it more complex as it will require defining new variables to suit each case.

Also, this is to keep current HaaS code operational. 

Once I am done with the client library, the CLI is going to be completely revamped.

**Comment by knikolla** [comment]
@SahilTikale yeah, I backtracked on my kwargs argument after I saw the `cmd` decorator.

File: haas/cli.py [hunk 1]
```diff
@@ -371,11 +371,49 @@ def switch_register(switch, subtype, *args):
...
+            return
+
+    else:
+        sys.stderr.write('ERROR: Wrong subtype supplied\n')
+        return
+
     url = object_url('switch', switch)
     do_put(url, data=switchinfo)
 
+
```
Comment (knikolla) [inline haas/cli.py]: extra blank line

**Comment by SahilTikale** [comment]
@knikolla  fixed it.

**Comment by knikolla** [comment]
+1

**Comment by kylehogan** [comment]
sorry for the late reply, I missed the initial email.
It looks good to me, your decision on how to handle the errors makes sense.

My only comment is that it might be helpful if the error for 'wrong subtype supplied' was instead either 'invalid subtype' or said that subtype must be 'nexus, mock, or powerconnect55xx' 
Listing all the subtypes might not make sense if there end up being many of them, but with only three it seems reasonable. This way it is clear that the subtype supplied is not one that is supported rather than 'wrong' which could imply that the subtype was valid, but a different subtype had been expected.

**Comment by knikolla** [comment]
@SahilTikale, I agree with Kyle. We are already having to handle all the switches in a separate code block. So we know what we support and can provide feedback to the user on the correct types.

**Comment by gsilvis** [comment]
This logic should be in the switch driver---that way it's extensible.

**Comment by gsilvis** [comment]
After some conversation with @SahilTikale, we agreed we don't need to do this nicely... yet.

However, please add a TODO or FIXME comment saying that this current design needs to be changed so that the CLI doesn't need to know about every backend.

**Comment by gsilvis** [comment]
(and once you do that, +1)

**Comment by gsilvis** [comment]
+1 and merging

**Pull request merged**